File: backend/Services/Database/Level.cs
using System.ComponentModel.DataAnnotations;

namespace Services.Database
{
    public class Level
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public int MaxPoints { get; set; }

        public int Order { get; set; }
    }
}

File: UI/justspeak_desktop/lib/models/level.dart
import 'package:json_annotation/json_annotation.dart';

part 'level.g.dart';

@JsonSerializable()
class Level {
  int id;
  String name;
  String description;
  int maxPoints;
  int order;

  Level({
    required this.id,
    required this.name,
    required this.description,
    required this.maxPoints,
    required this.order,
  });

  factory Level.fromJson(Map<String, dynamic> json) => _$LevelFromJson(json);
  Map<String, dynamic> toJson() => _$LevelToJson(this);
}
